From 43df6c543bd2ea8fa0ade12347a6fb93e82455ae Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 19 Apr 2016 18:20:29 +0300
Subject: [PATCH 0357/2241] mmc: a70x0: enable eMMC/SD for Armada70x0 in
 device-tree

- SD/eMMC working without card detect and without voltage change

Change-Id: I03116c7d55fdd43045ab3e3eb931973455b8e57b
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29214
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi    | 5 +++++
 arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi | 7 -------
 arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi | 3 ---
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      | 2 +-
 4 files changed, 6 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 11e63c8..1f6a06d 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -58,6 +58,11 @@
 
 	ap806 {
 		config-space {
+			sdhci@6e0000 {
+				bus-width = <4>;
+				non-removable; /* For now, need to configure the GPIO for card detect */
+				status = "okay";
+			};
 			serial@512000 {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
index b1b11395..95a1ff6 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
@@ -67,11 +67,4 @@
 			enable-method = "psci";
 		};
 	};
-	ap806 {
-		config-space {
-			sdhci@d8000 {
-				status = "okay";
-			};
-		};
-	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
index d3db017..3ff612e 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
@@ -82,9 +82,6 @@
 
 	ap806 {
 		config-space {
-			sdhci@d8000 {
-				status = "okay";
-			};
 			etm0@c40000 {
 				compatible = "arm,coresight-etm4x", "arm,primecell";
 				reg = <0xc40000 0x1000>;
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 700611e..3226add 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -239,7 +239,7 @@
 
 			};
 			/* The SDIO should not be enabled in Z1 DSTI/DST */
-			sdhci@d8000 {
+			sdhci@6e0000 {
 				compatible = "marvell,xenon-sdhci";
 				reg = <0x6E0000 0x300>;
 				interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
-- 
2.7.4

